aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1378,17 +1377,19 @@ TORCH_IMPL_FUNC(mean_out)
     // which, in turn, does not produce as accurate results.
     bool is_half_type = (dtype == kHalf || dtype == kBFloat16);
     auto sum_out_dtype = is_half_type ? ScalarType::Float : dtype;
-    result_mut = is_half_type ? result_mut.to(sum_out_dtype) : result_mut;
+    auto result_temp = is_half_type ? result.to(sum_out_dtype) : result;
```

Comment:
We'd better add other data types in the testing to ensure that they also return the correct data types.
@onlyCPU
@dtypes(torch.half, torch.bfloat16, torch.float, torch.double)
def test_mean_out_float16_is_alias_of_return(self, dtype, device):
a = torch.tensor([[[1.0, 1.0, 1.0, 1.0]], [[2.0, 2.0, 2.0, 2.0]], [[3.0, 3.0, 3.0, 3.0]]],
                         dtype=dtype, device=device)

Reply:
Thanks for your review. I add it.

Reply:
Thanks for the fix! Is there no way to avoid the copy?

Reply:
It cannot be avoided because the promotion needs more storage. Thus, it cannot reuse the storage of input "out" parameter and need to update the result to "out".

Reply:
I have added some comments about it. It is my first time do contribution to GitHub. Thanks for all you guy's excellent suggestions!